app/controllers/admin/campaigns_controller.rb

```diff
@@ -1,6 +1,7 @@
 class Admin::CampaignsController < AdminController
   before_action :set_campaign, only: [:show, :edit, :update, :destroy]
   helper_method :sort_column, :sort_direction
+  before_filter :authorize_admin
```

Comment:
I have done so, adding an exception to admin index for user role GUEST, to prevent infinite redirects :).